### PR TITLE
REMReM lookups controlled per lookup link instead of globally per call.

### PR DESCRIPTION
**Pull request by zvsddrg**
### Description of the Change
REMReM lookups controlled 'failIfNoneFound' and 'failIfMultipleFound' lookups per lookup object within an event instead of globally per call.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: wiki/markdown/statusCodes.md [hunk 1]
```diff
@@ -170,6 +170,7 @@ These response can be generated only when `/generateAndPublish` endpoint is used
 | 401         | Unauthorized          | Unauthorized. Please, check if LDAP for REMReM Generate Service is disabled | Is returned if LDAP for REMReM Generate Service is enabled and REMReM Generate Publish have not access to it.          |
 | 406         | Not Acceptable        | No event id found with ERLookup properties                                  | Is returned if no event id fetched from configured event repository in REMReM generate.                                |
 | 417         | Expectation Failed    | Multiple event ids found with ERLookup properties                           | Is returned if multiple event ids fetched from configured event repository in REMReM generate.                         |
+| 422         | Unprocessable Entity  | Link specific options could not fetch information from ER                   | Is returned if could not fetch information from ER with specific options.                                              |
```
Comment (e-backmark-ericsson) [inline wiki/markdown/statusCodes.md:173]: See my comment in the review of remrem generate

File: wiki/markdown/statusCodes.md [hunk 2]
```diff
@@ -230,6 +231,19 @@ The Lookup properties with multiple event ids fetched from configured event repo
 ]
 ```
 
+**422 Unprocessable Entity**
+The link specific options could not fetch information from configured event repository in generate , REMReM fails to generate.
```
Comment (e-backmark-ericsson) [inline wiki/markdown/statusCodes.md:235]: See my earlier comment

File: wiki/markdown/statusCodes.md [hunk 3]
```diff
@@ -230,6 +231,19 @@ The Lookup properties with multiple event ids fetched from configured event repo
...
+
+```
+[
+    {
+     "status_code": 422,
+     "result": "FAIL",
+     "message": "Link specific options could not fetch information from ER"
```
Comment (e-backmark-ericsson) [inline wiki/markdown/statusCodes.md:242]: See my earlier comment

File: wiki/markdown/statusCodes.md (hunk 1 above)
Comment (zvsddrg) [reply wiki/markdown/statusCodes.md]: changed error message as suggested

File: wiki/markdown/statusCodes.md (hunk 2 above)
Comment (zvsddrg) [reply wiki/markdown/statusCodes.md]: changed error message as suggested

File: wiki/markdown/statusCodes.md (hunk 3 above)
Comment (zvsddrg) [reply wiki/markdown/statusCodes.md]: changed error message as suggested

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 4]
```diff
@@ -175,10 +175,10 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         HttpEntity<String> entity = new HttpEntity<>(bodyJsonOut, headers);
-        EnumSet<HttpStatus> getStatus = EnumSet.of(HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNAUTHORIZED, HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.INTERNAL_SERVER_ERROR);
+        EnumSet<HttpStatus> getStatus = EnumSet.of(HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNAUTHORIZED, HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.UNPROCESSABLE_ENTITY);
 
         try {
-        	String generateUrl=generateURLTemplate.getUrl()+"&failIfMultipleFound="+failIfMultipleFound+"&failIfNoneFound="+failIfNoneFound+"&lookupInExternalERs="+lookupInExternalERs+"&lookupLimit="+lookupLimit;
+            String generateUrl=generateURLTemplate.getUrl()+"&failIfMultipleFound="+failIfMultipleFound+"&failIfNoneFound="+failIfNoneFound+"&lookupInExternalERs="+lookupInExternalERs+"&lookupLimit="+lookupLimit;
```
Comment (SantoshNC68) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:181]: Format this statement

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java [hunk 5]
```diff
@@ -164,4 +173,44 @@ public void testGenerateURLTemplate() throws Exception {
...
 
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testErLookupFailedWithOptions() throws Exception {
+        String correctURL = "/{mp}?msgType={msgType}";
+        when(restTemplate.postForEntity(Mockito.contains(correctURL), Mockito.<HttpEntity<String>>any(),
+                Mockito.eq(String.class), Mockito.anyMap())).thenReturn(responseOptionsFailed);
+
+        ResponseEntity<?> elem = unit.generateAndPublish("eiffelsemantics", "eiffelactivityfinished", "", "", "",false,
```
Comment (SantoshNC68) [inline publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java:184]: format

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java [hunk 6]
```diff
@@ -164,4 +173,44 @@ public void testGenerateURLTemplate() throws Exception {
...
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testErLookupFailedWithMultipleFound() throws Exception {
+        String correctURL = "/{mp}?msgType={msgType}";
+        when(restTemplate.postForEntity(Mockito.contains(correctURL), Mockito.<HttpEntity<String>>any(),
+                Mockito.eq(String.class), Mockito.anyMap())).thenReturn(responseMultipleFound);
+
+        ResponseEntity<?> elem = unit.generateAndPublish("eiffelsemantics", "eiffelactivityfinished", "", "", "",false,
```
Comment (SantoshNC68) [inline publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java:197]: format

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java [hunk 7]
```diff
@@ -164,4 +173,44 @@ public void testGenerateURLTemplate() throws Exception {
...
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testErLookupFailedWithNoneFound() throws Exception {
+        String correctURL = "/{mp}?msgType={msgType}";
+        when(restTemplate.postForEntity(Mockito.contains(correctURL), Mockito.<HttpEntity<String>>any(),
+                Mockito.eq(String.class), Mockito.anyMap())).thenReturn(responseNoneFound);
+
+        ResponseEntity<?> elem = unit.generateAndPublish("eiffelsemantics", "eiffelactivityfinished", "", "", "",false,
```
Comment (SantoshNC68) [inline publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java:210]: format

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java (hunk 4 above)
Comment (zvsddrg) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java]: Formatted this statement

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java (hunk 5 above)
Comment (zvsddrg) [reply publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java]: Done

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java (hunk 6 above)
Comment (zvsddrg) [reply publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java]: done

File: publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java (hunk 7 above)
Comment (zvsddrg) [reply publish-service/src/test/java/com/ericsson/eiffel/remrem/publish/service/EiffelRemremCommonControllerUnitTest.java]: done

**Pull request merged**